injector_test.py

```diff
@@ -1690,3 +1690,30 @@ def function1(a: int | str) -> None:
         pass
 
     assert get_bindings(function1) == {'a': Union[int, str]}
+
+
+# test for https://github.com/python-injector/injector/issues/217
+def test_annotated_instance_integration_works():
+    UserID = Annotated[int, "user_id"]
+
+    def configure(binder):
+        binder.bind(UserID, to=123)
+
+    injector = Injector([configure])
+    assert injector.get(UserID) == 123
+
+
+def test_annotated_class_integration():
+    class Shape(abc.ABC):
+        pass
+
+    class Circle(Shape):
+        pass
+
+    first = Annotated[Shape, "first"]
+
+    def configure(binder):
+        binder.bind(first, to=Circle)
+
+    injector = Injector([configure])
+    assert isinstance(injector.get(first), Circle)
```

Comment:
I'd like to see another test or two, with multiple bindings to see that Injector treats them as separate binds, for example:
- Making sure that `Annotated[int, "first"]` and `Annotated[int, "second"]` get separate bindings/values
- Making sure that `Annotated[int, "user_id"]` and `int` get separate bindings/values

What do you think?

Reply:
Thanks for the feedback! Makes sense, I will add some more tests and take a look at the failing ones.